common/workunit/workunit.cpp

```diff
@@ -331,7 +331,10 @@ class CConstGraphProgress : public CInterface, implements IConstWUGraphProgress
                 throwUnexpected();
             }
 
-            IPropertyTree * next = curTarget->addPropTree(tag);
+            VStringBuffer xpath("%s[@id=\"%s\"]", tag, id);
```

Comment:
Potentially less efficient to check if it exists already, but the "correct" solution is probably beyond 8.6.x